lib/Twig/Extensions/Node/Trans.php

```diff
@@ -41,6 +41,10 @@ public function compile(Twig_Compiler $compiler)
 
         $function = null === $this->getNode('plural') ? 'gettext' : 'ngettext';
 
+        if (null !== $this->getNode('domain')) {
```

Comment:
Allowing null doesn't make much sense. Passing null is the same as not passing it at all. Thus not passing a proper domain is an error condition IMO.